### PR TITLE
[TNG-75] ImportAnalysis page, based on ImportData page.

### PR DESCRIPTION
**Pull request by marikomedlock**
### Jira Ticket: https://precisionmedicineinitiative.atlassian.net/browse/TNG-75
This ticket is in the AoU JIRA, because Tanagra Data Explorer is part of the DRC.

### What
Per advice from the `terra-ui` team [in Slack](https://broadinstitute.slack.com/archives/C01EHNUM73R/p1690551236395189), I added a new `ImportAnalysis` page that is based heavily on the existing `ImportData` page.

Once this PR is live in dev, I will add a companion change to Tanagra Data Explorer to redirect to `ImportAnalysis` when the user wants to export a notebook file to Community Workbench Terra.

### Why
The Tanagra Data Explorer ([gh repo](https://github.com/DataBiosphere/tanagra/), [project overview](https://broad.io/tanagra)) can generate an ipynb notebook file with the embedded SQL for a user's cohort. We want to let users import this file to Community Workbench Terra directly. This is a request from the DRC VUMC team who will use the first deployment of Tanagra.

Here is the general user flow I have in mind:
- User clicks "export notebook file" button in Tanagra Data Explorer.
- Tanagra Data Explorer generates a signed URL to an ipynb file.
- Tanagra Data Explorer redirects to a Community Workbench Terra import page.
- User chooses a workspace and clicks "import" button on Community Workbench Terra import page.
- Community Workbench Terra copies the notebook file to the workspace bucket, notebooks folder.
- Community Workbench Terra redirects to the workspaces page, analyses tab.

### Testing strategy
I added unit tests for the new page.

### CODE REVIEW

File: src/libs/events.js [hunk 1]
```diff
@@ -20,6 +20,7 @@ const eventsList = {
   analysisEnableBeta: 'analysis:enable',
   analysisDisableBeta: 'analysis:disable',
   analysisCreate: 'analysis:create',
+  analysisImport: 'analysis:import',
```
Comment (marikomedlock) [inline src/libs/events.js:23]: I added a new event here, and see in the wiki that I'll also need to add it to the mixpanel lexicon. I should do this *after* merging the PR, is that correct?

File: src/pages/ImportAnalysis.js [hunk 2]
```diff
@@ -0,0 +1,419 @@
+import _ from 'lodash/fp';
```
Comment (jdcanas) [inline src/pages/ImportAnalysis.js]: is there any reason not to introduce this new file as a typescript file? Seems pretty unsafe to introduce a javascript file when a lot of the underlying imports would be safer consumed with type checkcing

**Review by jdcanas — CHANGES_REQUESTED** [review]
Pretty against introducing new large files not as `.ts` files without a good reason

File: src/pages/ImportAnalysis.js [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+    return false;
+  }
+  return !!workspace.workspace.bucketName && workspace.workspace.bucketName.startsWith('fc-secure');
+};
+
+// This method fetches the raw file from the URL, then writes it to the
+// workspace bucket, notebooks folder.
+export const importAnalysisFile = (namespace, name, url, filename, runtime) => {
+  return async () => {
+    if (!isRuntimeToolLabel(runtime)) {
```
Comment (jdcanas) [inline src/pages/ImportAnalysis.js]: this seems pretty confusing. This method in particular would benefit from types. I'm guessing this variable `runtime` is actually of type `ToolLabel` and not `Runtime`. The error message below probably needs changing as a result.

**Comment by calypsomatic** [comment]
@marikomedlock Just came across this PR and see that it is very old.  Is this still active/valid?

**Comment by marikomedlock** [comment]
> @marikomedlock Just came across this PR and see that it is very old. Is this still active/valid?

No, product decided not to pursue this feature.

**Pull request closed without merging**